tests/test_submit_tuxbuild.py

```diff
@@ -52,243 +63,88 @@ def submit_tuxbuild(self, tuxbuild):
-            'git_sha': "5302568121ba345f5c22528aefd72d775f25221e",
-            'git_short_log': '5302568121ba ("Add linux-next specific files for 20201021")',
-            'git_describe': "next-20201021",
-            'kconfig': [base_kconfig + ["CONFIG_ARM64_MODULE_PLTS=y"], base_kconfig + ["CONFIG_IGB=y", "CONFIG_UNWINDER_FRAME_POINTER=y"]],
-            'git_branch': os.environ.get("KERNEL_BRANCH"),
-            'make_kernelversion': "5.9.0",
-            'kernel_version': "5.9.0",
-            'config': configs,
-            'download_url': urls,
-        }
-        for expected_key in expected_metadata.keys():
-            self.assertEqual(expected_metadata[expected_key], getattr(build.metadata, expected_key))
+        group = self.squad.group("my_group")
+        self.assertIsNotNone(group)
 
-        # Make sure there's no extra attributes in the metadata object
-        metadata_attrs = build.metadata.__dict__
-        del metadata_attrs["id"]
-        self.assertEqual(sorted(expected_metadata.keys()), sorted(metadata_attrs.keys()))
-
-        # Check results for v4.4.4, which has 1 instance in build.json
-        build = project.build("v4.4.4")
-        # Make sure metadata values match expected values
-        url = 'https://builds.tuxbuild.com/%s/' % 'B3TECkH4_1X9yKoWOPIhew'
-        config = url + "config"
-        expected_metadata = {
-            'git_repo': "https://gitlab.com/Linaro/lkft/mirrors/next/linux-next",
-            'git_ref': None,
-            'git_commit': "5302568121ba345f5c22528aefd72d775f25221e",
-            'git_sha': "5302568121ba345f5c22528aefd72d775f25221e",
-            'git_short_log': '5302568121ba ("Add linux-next specific files for 20201021")',
-            'git_describe': "v4.4.4",
-            'kconfig': base_kconfig + ["CONFIG_ARM64_MODULE_PLTS=y"],
-            'git_branch': os.environ.get("KERNEL_BRANCH"),
-            'make_kernelversion': "5.9.0",
-            'kernel_version': "5.9.0",
-            'config': config,
-            'download_url': url,
-        }
-        for expected_key in expected_metadata.keys():
-            self.assertEqual(expected_metadata[expected_key], getattr(build.metadata, expected_key))
+        project = group.project("my_project")
+        self.assertIsNotNone(project)
 
-        # Make sure there's no extra attributes in the metadata object
-        metadata_attrs = build.metadata.__dict__
-        del metadata_attrs["id"]
-        self.assertEqual(sorted(expected_metadata.keys()), sorted(metadata_attrs.keys()))
+        build = project.build("next-20211224")
+        self.assertIsNotNone(build)
 
-        for arch in ["arm64", "x86"]:
-            environment = (
-                self.squad.group("my_group").project("my_project").environment(arch)
-            )
-            self.assertIsNotNone(environment, "environment %s does not exist" % (arch))
+        for m in squad_client.tux.ALLOWED_METADATA:
+            self.assertTrue(hasattr(build.metadata, m))
```

Comment:
I'd still check all `build.metadata` values. It's useful to check what to expect instead of having to open build.json

Reply:
@chaws Can you provide an example of what you mean? I tried to remove a bunch of testing that I wasn't sure added any real value, so maybe I went overboard.

Reply:
If you check only the keys, values might be wrong. That was the case when I created this PR https://github.com/Linaro/squad-client/pull/137

That bug messed up metadata, so you couldn't be sure that the download_url for a given build in qa-reports was actually the correct one.